src/components/ColorPicker.tsx

```diff
@@ -42,14 +49,90 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThoughtId = useSelector(state => !!state.cursor && head(state.cursor))
+  const currentThoughtValue = useSelector(
+    state => (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || '',
+  )
+  const simplePath = useSelector(state => simplifyPath(state, state.cursor!))
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+
+  const selected = useMemo(() => {
+    // Define the color and background color regex to get the current color of current thought
+    const colorRegex = /color="(#[0-9a-fA-F]{6})"/g
+    const bgColorRegex = /background-color:\s*(rgb\(\d{1,3},\s?\d{1,3},\s?\d{1,3}\))/g
```

Comment:
`color` uses hex and `background-color` uses rgb. Why is that? If there are implicit dependencies here, they need to be made explicit in a comment. It's not obvious from just looking at the code.